### ISSUE TITLE
Critical: Checkbox keeps set even if unset/unchecked

### ISSUE THREAD

**Issue by mvorisek** [atk4/ui#902]
Test code:
```
<?php

require __DIR__ . '/vendor/autoload.php';

$app = new \atk4\ui\App('My App');
$app->initLayout('Admin');

/* @var $form \atk4\ui\Form */
$form = $app->add('Form');
$form->addField('purge_missing', ['CheckBox']);
$form->buttonSave->set('Import');

$form->onSubmit(function(\atk4\ui\Form $form) {
    throw new \Exception($form->model['purge_missing']);
});

```

If the form is just send, the `$form->model['purge_missing']` evaluates to `0`. Fine.

But if the checkbox is checked, form is sent, and then unchecked and submitted again, the value is read as `1`!

Please reproduce and fix asap.

**Comment by ibelar** [comment]
@mvorisek - Sorry could not reproduce the issue with your code above.  Submitting form multiple time always send proper data, whether checkbox input is check or unchecked.

**Comment by mvorisek** [comment]
I have updated the example - the issue occur when an exception is rendered on the server side (and for some reasons then the `purge_missing` value after check/uncheck is sent to the server as `true`).

![image](https://user-images.githubusercontent.com/2228672/73700272-63435180-46e6-11ea-8c5a-4b1f795f7d1b.png)

@ibelar can you please retest the updated example?

**Comment by ibelar** [comment]
@mvorisek - Thanks for the update. 
Fix in  #909

**Comment by mvorisek** [comment]
@ibelar Great job!
